### PR TITLE
[gitlab] Sign staging deb repodata with the same key as we use to sign deb packages

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

Makes deb-s3 use the same key to sign staging repo repodata as is used to sign individual deb packages.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

As soon as the first RC of the next release is uploaded, https://s3.amazonaws.com/apt.datad0g.com/dists/beta/Release.gpg should be a correct GPG signature of https://s3.amazonaws.com/apt.datad0g.com/dists/beta/Release using key https://s3.amazonaws.com/public-signing-keys/DATADOG_APT_KEY_382E94DE.public and it should be possible to install packages from the repository.

### CODE REVIEW

**Review by albertvaka — APPROVED** [review]
LGTM. Can we get a second pair of eyes on it before merging?

**Review by KSerrania — APPROVED** [review]
LGTM. About your testing note, we can check this even before the first RC, as the next `nightly` build should sign the `nightly` channel with this key.

**Pull request merged**
